DemoNativeControls/app/src/main/java/com/jwplayer/demo/nativecontrols/JWPlayerNativeControls.java

```diff
@@ -0,0 +1,253 @@
+        mRewindButton.setOnClickListener(this);
+        mFullScreenButton.setOnClickListener(this);
+        mMuteButton.setOnClickListener(this);
+
+        mVideoProgressBar = new JWProgressBar();
+        mSeekBar.setOnSeekBarChangeListener(mVideoProgressBar);
+        resetVideoPlaybackUI();
+    }
+
+    private void resetVideoPlaybackUI(){
+        mPlayPauseButton.setImageResource(R.drawable.exo_controls_play);
+        mSeekBar.setProgress(0);
+        mSeekBarText.setText("00:00 / 00:00");
+    }
+
+    public void addControlsInteractionListener(OnControlsInteraction ocl){
+        mControlChangeListeners.add(ocl);
+    }
+
+    public boolean removeControlsInteractionListener(OnControlsInteraction ocl){
+        return mControlChangeListeners.remove(ocl);
+    }
+
+    public void setJWView(JWPlayerView playerView) {
+        mPlayerView = playerView;
+        //Sign us up for player related events so we can appopriately update UI
+        //Play and pause will be used to toggle the play/pause button and it's functionality
+        mPlayerView.addOnPlayListener(this);
+        mPlayerView.addOnPauseListener(this);
+        //We'll use time event to update the seek bar time
+        mPlayerView.addOnTimeListener(mVideoProgressBar);
+        //Listen to the complete event to update the play pause UI
+        mPlayerView.addOnCompleteListener(this);
+    }
+
+    public void unSubscribeFromJWEvents(){
+        mPlayerView.removeOnPlayListener(this);
+        mPlayerView.removeOnPauseListener(this);
+        mPlayerView.removeOnTimeListener(mVideoProgressBar);
+        mPlayerView.removeOnCompleteListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        boolean stoppedPlayback = false;
+        switch (v.getId()) {
+            case R.id.play_and_pause_button:
+                //If we're playing stop us, if we're not playing play us
+                if(mPlayerView.getState() == PlayerState.PLAYING){
+                    mPlayerView.pause();
+                    stoppedPlayback = true;
+                } else if(mPlayerView.getState() != PlayerState.PLAYING){
+                    mPlayerView.play();
+                }
+                break;
+            case R.id.stop_button:
+                //Cancel playback ASAP and reset the UI back to the pre video starts
+                mPlayerView.stop();
+                resetVideoPlaybackUI();
+                break;
+            case R.id.fullscreen_button:
+                if(mPlayerView.getFullscreen()){
+                    mPlayerView.setFullscreen(false,true);
+                } else {
+                    mPlayerView.setFullscreen(true,true);
+                }
+                break;
+            case R.id.fast_forward_button:
+                mPlayerView.seek(mPlayerView.getPosition() + 10);
+                break;
+            case R.id.rewind_button:
+                double newPosition = mPlayerView.getPosition() - 10;
```

Comment:
Have you tested this with Live streams?

Reply:
I do after this comment.
There's now a button in the hamburger menu up top to switch to a stream layout.